### PR TITLE
PROD-256/implement-revealclaim-dialog-from-campaigndetail-page

### PR DESCRIPTION
**Pull request by danijel-radomiljac**
@Yihwan  I connected just functionality to reveal and claim. Still waiting for new dialog to be merged. 

One question, where should "View results" redirect user?

### CODE REVIEW

File: app/layout.tsx [hunk 1]
```diff
@@ -30,13 +34,19 @@ export const metadata: Metadata = {
...
 
-export default function RootLayout({
+export default async function RootLayout({
   children,
 }: Readonly<{
   children: React.ReactNode;
 }>) {
   const isDemo = process.env.ENVIRONMENT === "demo";
 
+  const user = await getCurrentUser();
```
Comment (marvinmarnold) [inline app/layout.tsx:44]: Any thoughts about how this impacts load time? Does any extra  loading behavior like `<Suspense>` need to be added anywhere?

**Review by marvinmarnold — COMMENTED** [review]
How should I be testing? When I go to [campaigns page](https://chomp-git-prod-193-enable-navbar-dailydeck-gator-labs.vercel.app/campaigns), I can't click on anything, in order to lead to reveal flow.
Screenshot 2024-09-11 at 12 26 15 PM

**Comment by danijel-radomiljac** [comment]
> How should I be testing? When I go to [campaigns page](https://chomp-git-prod-193-enable-navbar-dailydeck-gator-labs.vercel.app/campaigns), I can't click on anything, in order to lead to reveal flow. Screenshot 2024-09-11 at 12 26 15 PM

Log in to test it

**Comment by marvinmarnold** [comment]
I am logged in
https://github.com/user-attachments/assets/f18d92e9-2698-4f18-bde4-f574cbbde379

**Comment by danijel-radomiljac** [comment]
> I am logged in. Screen recording: [user-attachments/assets/f18d92e9-2698-4f18-bde4-f574cbbde379](https://github.com/user-attachments/assets/f18d92e9-2698-4f18-bde4-f574cbbde379)

You are not in my preview link. https://chomp-git-prod-256-implement-revealclaim-dial-2d05bd-gator-labs.vercel.app/login?next=%2Fapplication%3FvercelToolbarCode%3DZnt7hofA8gFXgeh

**Comment by marvinmarnold** [comment]
> You are not in my preview link
My bad, thanks.

I'm going to comment in Linear, where it's easier to add screen recordings.

**Review by marvinmarnold — COMMENTED** [review]
Am I in the wrong Figma? I'm seeing different designs.
Screenshot 2024-09-11 at 1 41 17 PM

Can you make cursor a pointer on hover, when selecting a deck to reveal?
Screenshot 2024-09-11 at 1 39 57 PM

**Comment by danijel-radomiljac** [comment]
> Am I in the wrong Figma? I'm seeing different designs. Screenshot 2024-09-11 at 1 41 17 PM
> 
> Can you make cursor a pointer on hover, when selecting a deck to reveal? Screenshot 2024-09-11 at 1 39 57 PM

As I wrote in PR description, I am waiting for dialog PR to be merged: https://github.com/gator-labs/chomp/pull/482

I just added funcionality claim deck / reveal deck

File: app/layout.tsx (hunk 1 above)
Comment (danijel-radomiljac) [reply app/layout.tsx:44]: It is simple get request, it should not impect that much load time

**Comment by Yihwan** [comment]
since this PR depends on some others, i think it would be best to handle design review end-to-end when more components are merged into prod. so will defer cr to @marvinmarnold

**Comment by marvinmarnold** [comment]
@danijel-radomiljac to review [Dialog](https://github.com/gator-labs/chomp/pull/482) component and finish up this PR

**Comment by Yihwan** [comment]
we will likely need to remove the "Reveal Results" state, but this can be handled as part of follow up tasks

**Comment by Yihwan** [comment]
merging this; any follow up from cr can be separate pr and can revert this one if needed

**Pull request merged**